aspnetcore/signalr/javascript-client.md

```diff
@@ -98,7 +98,140 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 ## Reconnect clients
 
-The JavaScript client for SignalR doesn't automatically reconnect. You must write code that will reconnect your client manually. The following code demonstrates a typical reconnection approach:
+::: moniker range=">= aspnetcore-3.0"
+
+### Automatically reconnect
+
+The JavaScript client for SignalR can be configured to automatically reconnect using the [withAutomaticReconnect](/javascript/api/%40aspnet/signalr/hubconnectionbuilder#withautomaticreconnect) method on [HubConnectionBuilder](/javascript/api/%40aspnet/signalr/hubconnectionbuilder). It will not automatically reconnect by default.
+
+```javascript
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("/chatHub")
+    .withAutomaticReconnect()
+    .build();
+```
+
+Without any parameters, `withAutomaticReconnect()` configures the client to wait 0, 2, 10 and 30 seconds respectively before trying up to 4 reconnect attempts.
+
+Before starting any reconnect attempts, the `HubConnection` will transition to the [Reconnecting](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onreconnecting](/javascript/api/%40aspnet/signalr/hubconnection#onreconnecting) callbacks instead of transitioning to the [Disconnected](javascript/api/%40aspnet/signalr/hubconnectionstate#disconnected) state and triggering its [onclosed](/javascript/api/%40aspnet/signalr/hubconnection#onclosed) callbacks like a `HubConnection` without automatic reconnect configured. This provides an opportunity to to warn users that the connection has been lost and disable UI elements.
```

Comment:
```suggestion
Before starting any reconnect attempts, the `HubConnection` will transition to the [Reconnecting](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onreconnecting](/javascript/api/%40aspnet/signalr/hubconnection#onreconnecting) callbacks instead of transitioning to the [Disconnected](javascript/api/%40aspnet/signalr/hubconnectionstate#disconnected) state and triggering its [onclosed](/javascript/api/%40aspnet/signalr/hubconnection#onclosed) callbacks like a `HubConnection` without automatic reconnect configured. This provides an opportunity to warn users that the connection has been lost and to disable UI elements.
```